### ISSUE TITLE
Is there function to execute a shell command

### ISSUE THREAD

**Issue by jarnoshen** [matz/streem#136]
As a shell script, is there a function to execute a shell command (e.g. "ls", "pwd", etc.), already?

**Comment by matz** [comment]
Sorry, not yet.

**Comment by mattn** [comment]
it's easy to implement with popen, i guess.

**Comment by amiralles** [comment]
Hey @matz! Big fan of your work!

Would the syntax for this feature look something like this?
```
system('ls') { in, out, err -> ... } | stdout
```
If so, I could try to implement it. Thanks!

**Comment by matz** [comment]
What I thought was:
```
spawn("ls") | stdout                           # can be a producer
stream | spawn("grep foo") | stdout  # and also be a filter
stream | spawn("cat > /tmp/a")         # and a consumer
```

**Comment by amiralles** [comment]
Yeah, that's way better. I'll explore that route. Thanks for replying!

**Comment by amiralles** [comment]
I managed to implement the first use-case on my fork (and it seems to work fine so far)
```
spawn("ls") | stdout    # can be a producer
```

Now I'm stuck at this one:
```
stream | spawn("grep foo") | stdout  # and also be a filter
```

As far as I understand, a stream can be a producer, a consumer, or a filter, but not all at the same time. So, the first question is: 

Is it possible for the *spawn* function to produce different kinds of streams based on their usage? 

If so, is there a way to identify how the function is being used (maybe based on the stream parameter or something like that)?

Thanks!
